### PR TITLE
Fix for ResourceConverter class. 

### PR DESCRIPTION
**Pull request by cosmin-ciuc**
When executing concurrently multiple asynchronous requests that return a rather large set of data the ResourceConverter.WriteJson() method needs to use a local copy of the JsonSerializer object because this method modifies the Converters collection of the JsonSerializer object.

The JsonHalMediaTypeOutputFormatter reuses the JsonSerializer across multiple concurrent requests.

I added class WebApi.Hal.Web.Api.Resources.BeerDetailListRepresentation for testing purposes. Also for testing purposes I added method 
```
[HttpGet("largeset")]
public async Task<BeerDetailListRepresentation> GetLargeSet(int setSize = 500)
```
in WebApi.Hal.Web.Api.BeerDetailController class.

### CODE REVIEW

File: WebApi.Hal.Web/Startup.cs [hunk 1]
```diff
@@ -46,18 +65,12 @@ public void Configure(IApplicationBuilder app, IHostingEnvironment env, ILoggerF
                 app.UseDeveloperExceptionPage();
             }
 
-            app.UseMvc();
+            beerDbContext.Database.Migrate();
```
Comment (panmanphil) [inline WebApi.Hal.Web/Startup.cs]: This line blew up for me with a table already exists error. Could you put a test so it only runs if it exists?

Comment (cosmin-ciuc) [reply WebApi.Hal.Web/Startup.cs]: Sorry, I'm not sure I understand you correctly. Are you saying that trying to execute beerDbContext.Database.Migrate(); on an existing database failed?

**Comment by cosmin-ciuc** [comment]
> Without actually trying this with some of my existing code, it looks like it would be drop in compatible unless you were working code specifically with input and output formatters. Do you agree?

I didn't understood your question. Are you saying that the code in this branch doesn't work unless you are explicitly using OutputFormatters? To be honest I don't see why somebody would use this library without OutputFormatters. This library is essentially an OutputFormatter, right?

Do you want to add some tests for null? If so, please say where.

Thank you.

**Comment by panmanphil** [comment]
Correct it fails on an existing database file with the table in it. i don't
recall if the db is a tracked file so it may have been a db from a previous
run, but the code worked with what I had for a file after commenting out
that line. The alternatives would be to check for the file and the table in
the file and skip, or create the file and initialize the db if not found?

I've barely used that demo site! But running it I see a number of important
concepts missing in the representation, especially curies. Ah of only there
was more time in the day.

**Comment by cosmin-ciuc** [comment]
Thank you for clarifying this aspect. If I remember correctly, I have encountered the same error ("table already exists") when I tried to run the new code with a database created by the old version of the code. Probably, in order to quickly surpass this problem, I have deleted the old database and then my code created it at the next run.
It might be the case that the old database is not compatible with EF Core 2.1 Code-First. Probably the old database does not have the __EFMigrationsHistory table and therefore EF Core 2.1 assumes that none of the table exists and tries to create them.
Taking into account that this is just a demo web application meant to be used to familiarize yourself with the using of this library, do you think is worth investing effort into making sure that the database is correctly migrated from the old version to the new version. I mean, would it be too much to just say: "If you have a database created using the demo application from before MM/DD/YYYY please delete it before running the current demo application."?

**Comment by cosmin-ciuc** [comment]
Yes, I'm 99% sure that repeatedly running the new demo app it is possible and the call to MigrateDb won't crash the app if the db already exists. The only trick is that you have to make sure you don't try to run the new app on a database created by the  old demo app.
Anyway, I'll test tonight that you can repeatedly run the new app without deleting the database. 
What about your other question, about the input and output formatters? I have to be honest and say I didn't understood it. Are you concerned that the my branch won't work in a setup where you don't have input or output formatter? If that's the case, rest assured, it works in the particular case when you don't use input formatter.
